File: Rpsls.java
package rpsls;

import java.util.Random;

/**
 *
 * @author vinoth
 */
public class Rpsls {

    static String number_to_name(int n) {
        
        switch(n) {
            case 0:
                return "rock";
            case 1:
                return "Spock";
            case 2:
                return "paper";
            case 3:
                return "lizard";
            case 4:
                return "scissors";
            default:
                return "Not defined";
        }
    }
    
    static int name_to_number(String name) {
        
        switch(name) {
            case "rock":
                return 0;
            case "Spock":
                return 1;
            case "paper":
                return 2;
            case "lizard":
                return 3;
            case "scissors":
                return 4;
            default:
                return 5;
        }
    }
    
    static void winner(String player_guess){
        
        Random rand = new Random();
        
        int player_number = name_to_number(player_guess);

        int comp_number = rand.nextInt(5);

        String comp_name = number_to_name(comp_number);

        System.out.println("Player chooses " + player_guess);
        System.out.println("Computer chooses " + comp_name);

        if (comp_number == player_number) {
            System.out.println("Player and computer tie!\n");
        } else if ((comp_number - player_number) % 5 < 3) {
            System.out.println("Computer wins!\n");
        } else {
            System.out.println("Player wins!\n");
        }
    }
    
    public static void main(String[] args) {
        
        // Test samples
        winner("rock");
        winner("Spock");
        winner("paper");
        winner("lizard");
        winner("scissors");
    }
}

File: rpsls.py
# Rock Paper Scissors Lizard Spock

# to import random function for using random.randrange
import random

# function to convert number to name
def number_to_name(number):
    if number == 0:
        return "rock"
    elif number == 1:
        return "Spock"
    elif number == 2:
        return "paper"
    elif number == 3:
        return "lizard"
    elif number == 4:
        return "scissors"
    else:
        return "Error"

# function to convert name to number
def name_to_number(name):
    if name == "rock":
        return 0
    elif name == "Spock":
        return 1
    elif name == "paper":
        return 2
    elif name == "lizard":
        return 3
    elif name == "scissors":
        return 4
    else:
        print name + "is not a character in RPSLS"


# function which selects the winner
def rpsls(guess): 
    
    # convert name to player_number using name_to_number
    player_number = name_to_number(guess)
    
    # compute random guess for comp_number using random.randrange()
    comp_number = random.randrange(0,5)

    # compute difference of player_number and comp_number modulo five
    winner = (comp_number - player_number) % 5

    # use if/elif/else to determine winner
    if winner < 3:
        player_win = False
    else:
        player_win = True
    
    # convert comp_number to name using number_to_name
    comp_name = number_to_name(comp_number)
    
    # print results
    print "Player chooses " + guess
    print "Computer chooses " + comp_name
    if player_win:
        print "Player wins!\n"
    elif comp_number == player_number:
        print "Player and computer tie!\n"
    else:
        print "Computer wins!\n"

    
player_guess = raw_input('Enter your choice.. Options: rock, Spock, paper, lizard, scissors')

rpsls(str(player_guess))